lib/HALSim/IMU.cpp

```diff
@@ -0,0 +1,187 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  IMU implementation for the Simulation
+ * @author Juliane Kerpe <juliane.kerpe@web.de>
+ */
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include "IMU.h"
+#include <limits>
+/******************************************************************************
+ * Compiler Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and classes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Prototypes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Local Variables
+ *****************************************************************************/
+
+/******************************************************************************
+ * Public Methods
+ *****************************************************************************/
+
+bool IMU::init()
+{
+    bool isInitializationSuccessful = false;
+    if ((nullptr != m_accelerometer) && (nullptr != m_gyro) && (nullptr != m_magnetometer))
+    {
+        m_accelerometer->enable(m_simTime.getTimeStep());
+        m_gyro->enable(m_simTime.getTimeStep());
+        m_magnetometer->enable(m_simTime.getTimeStep());
+        isInitializationSuccessful = true;
+    }
+    return isInitializationSuccessful;
+}
+
+void IMU::readAccelerometer()
+{
+    if (nullptr != m_accelerometer)
+    {
+        const double* accelerationValues = m_accelerometer->getValues();
+        if (nullptr != accelerationValues)
+        {
+            m_accelerationValues.valueX = convertFromDoubleToInt16(accelerationValues[0]); /** Index 0 is the x-axis. */
+            m_accelerationValues.valueY = convertFromDoubleToInt16(accelerationValues[1]); /** Index 1 is the y-axis. */
```

Comment:
Doxygen comment inside code will destroy the generated docu. Remove one * in the postfix.

Reply:
This comment is valid for other locations in the code too.